core/impl/src/main/java/com/blazebit/persistence/impl/function/datediff/week/DB2WeekDiffFunction.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2014 - 2019 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.impl.function.datediff.week;
+
+/**
+ * @author Jan-Willem Gmelig Meyling
+ * @since 1.4.0
+ */
+public class DB2WeekDiffFunction extends WeekDiffFunction {
+
+    public DB2WeekDiffFunction() {
+        // NOTE: we need lateral, otherwise the alias will be lost in the subquery
+        super("(select sign(i1) * floor(abs(i1)) from lateral(values (trunc(-(cast(?1 as date) - cast(?2 as date))) / 7)) as temp(i1))");
```

Comment:
Shouldn't a cast to int be enough?

Reply:
Also, casting to date here will result in loss of precision

Reply:
Db2, psql and oracle round to the closest int when casting, which is not appropiate here. Ill try to find another impl for this, i copied the days extraction from the db2 day diff function. Anyways, the db2 tests currently struggle anyway.